daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/unparsers/runtime1/ChoiceAndOtherVariousUnparsers.scala

```diff
@@ -26,14 +28,14 @@ import org.apache.daffodil.runtime1.processors._
 import org.apache.daffodil.runtime1.processors.unparsers._
 
 case class ChoiceBranchMap(
-  lookupTable: Map[ChoiceBranchEvent, Unparser],
+  lookupTable: java.util.Map[ChoiceBranchEvent, Unparser],
```

Comment:
Suggest centralizing all of this around this type definition
which you then put all the scaladoc/comment documentation on it. 
```
type ProperlySerializableMap = java.util.Map
```
Then use ProperlySerializableMap in all the places you currently java.util.Map explicitly.